lib/components/SelectionControl/radio/radio-button/index.tsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import clsx from "clsx";
+import {
+    LabelPairedCircleMdRegularIcon,
+    LabelPairedCircleDotMdFillIcon,
+    LabelPairedCircleInfoMdRegularIcon,
+} from "@deriv/quill-icons";
+import { Text } from "../../../Typography";
+import "./radio-button.scss";
+
+type TRadio = {
+    className?: string;
+    classNameInfo?: string;
+    classNameLabel?: string;
+    defaultChecked?: boolean;
+    disabled?: boolean;
+    has_info?: boolean;
+    id?: string;
+    name?: string;
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    required?: boolean;
+    value?: string;
+};
+
+const RadioButton = ({
+    children,
+    className,
+    classNameInfo,
+    classNameLabel,
+    defaultChecked,
+    disabled = false,
+    has_info,
+    id,
+    onChange,
+    value,
+    ...otherProps
+}: React.PropsWithChildren<TRadio>) => {
+    const [checked, setChecked] = React.useState(defaultChecked);
+
+    React.useEffect(() => {
+        setChecked(defaultChecked);
+    }, [defaultChecked]);
+
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setChecked(e.target.checked);
+        onChange && onChange(e);
+    };
+
+    const handleIconClick = (e: React.MouseEvent<HTMLSpanElement>) => {
+        if (!disabled) {
+            const synthesizedEvent = new Event("change", { bubbles: true });
+            const input = e.currentTarget.previousSibling as HTMLInputElement;
+            input.dispatchEvent(synthesizedEvent);
+        }
+    };
+
+    return (
+        <div>
+            <label
+                htmlFor={id}
+                className={clsx("quill-radio-button", className, {
+                    "quill-radio-group__item--active": checked,
+                })}
+                data-testid="dt_quill_radio_button"
+            >
+                <input
+                    className="quill-radio-button__input"
+                    type="radio"
+                    onChange={onInputChange}
```

Comment:
Should it work with onKeyDown too? onFocus

Reply:
It shouldn't matter on this input, it is hidden itself, but let me check on the actual button.

Reply:
I mean not input itself, but the visual effect, that it can be focused (e.g. around icon)